app/graphql/types/concerns/profile_scoring.rb

```diff
@@ -31,8 +32,16 @@ def total_host_count
     def test_result_host_count
       ::CollectionLoader.for(
         object.class,
-        object.policy_id ? :policy_test_result_hosts : :test_result_hosts
-      ).load(object).then(&:count)
+        test_result_method
+      ).load(object).then do |test_results|
+        Host.where(id: test_results.latest.supported.select(:host_id)).count
```

Comment:
I'm not sure whether not counting unsupported here is correct. Logically I'd expect `testResultHostCount` return number of host from all latest test results, regardless of supportability. Consumer can construct the number by subtracting `unsupportedHostCount`.

Reply:
I guess this gets to @bastilian's point about handling things in the backend as much as possible. What do you think here, @bastilian?

The way I see it, `compliantHostCount` and `totalHostCount` should never include unsupported hosts, even from the backend, since that's what the jira means by 'Report scoring is based only on systems using supported SSG versions'. When we implement this same thing in the REST API, it will also need to report counts without unsupported systems included, no UI logic involved.

Reply:
It would redefine meaning of "total". I see this as a possible API breakage. `supportedHostCount` would be more accurate.

Reply:
Exactly. Total now refers only to systems with supported uploads - that's the feature.

If we add `supportedHostCount`, we'd have to update all frontend uses of `totalHostCount` and change it to `supportedHostCount`, at which point we could remove `totalHostCount` completely (and you could claim we would break the API by doing so).

I don't think your point is invalid (especially wrt explicit naming), I just think that we are supposed to change the meaning of 'total' for this feature when it comes to policy scoring.

Reply:
Oh, apologies, this is `testResultHostCount` not the `totalHostCount`. Sorry, it was too many changes to review today.

Reply:
I also didn't realize that :)